src/cljs/athens/db.cljs

```diff
@@ -525,44 +525,24 @@
 ;; history
 
 (defonce history (atom []))
-(def ^:const history-limit 10)
-
-
-(defn drop-tail
-  [xs pred]
-  (loop [acc []
-         xs  xs]
-    (let [x (first xs)]
-      (cond
-        (nil? x) acc
-        (pred x) (conj acc x)
-        :else  (recur (conj acc x) (next xs))))))
-
-
-(defn trim-head
-  [xs n]
-  (vec (drop (- (count xs) n) xs)))
-
-
-(defn find-prev
-  [xs pred]
-  (last (take-while #(not (pred %)) xs)))
-
-
-(defn find-next
-  [xs pred]
-  (fnext (drop-while #(not (pred %)) xs)))
-
+#_(def ^:const history-limit 10)
 
+;; this gives us customization options
+;; now if there is a pattern for a tx then the datoms can be
+;; easily modified(mind the order of datoms) to add a custom undo/redo strategy
+;; Not seeing a use case now, but there is an option to do it
```

Comment:
Custom undo/redo logic: undo tree?

- https://github.com/mbbill/undotree
- https://www.emacswiki.org/emacs/UndoTree

Reply:
I've imagined a scenario where an op can be done in one step but undone in two steps or there is an explicit confirm action warning mid-step. 

Undotree looks very cool. A very power-user feature.

Reply:
Yeah, I've never personally used it either in Vim or Emacs, probably why it's a plugin. Though undotree is basically a visualization for versioning (at the block level).

This is Version Story, a startup in the current YC batch. I guess what they found is that versioning is really important for legal docs: https://www.versionstory.com. 

Git Tree viz is a similar concept.